community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCursor.java

```diff
@@ -776,4 +776,20 @@ public void setCursorException( String message )
             this.cursorException = message;
         }
     }
+
+    @Override
+    public void clear()
+    {
+        if ( pageSize == 0 )
+        {
+            // if this page has been closed then pageSize == 0 and we must adhere to making writes
+            // trigger outOfBounds when closed
+            outOfBounds = true;
+        }
+        else
+        {
+            UnsafeUtil.setMemory( pointer, pageSize, (byte) 0 );
+            offset = 0;
```

Comment:
It felt relevant, perhaps it isn't. For the driving use case it isn't necessary, but zapping it really feels like a `reset --hard`, so to speak, which sort of implies offset reset as well. Doesn't matter to me.